src/Core/Record.php

```diff
@@ -79,112 +74,86 @@ public function __construct($xml)
         }
     }
 
-    /**
-     * @return string
-     */
-    public function getRecordId()
+    public function getRecordId(): string
     {
         return $this->recordId;
     }
 
-    /**
-     * @return string
-     */
-    public function getMeetingId()
+    public function getMeetingId(): string
     {
         return $this->meetingId;
     }
 
-    /**
-     * @return string
-     */
-    public function getName()
+    public function getName(): string
     {
         return $this->name;
     }
 
-    /**
-     * @return null|bool
-     */
-    public function isPublished()
+    public function isPublished(): ?bool
     {
         return $this->isPublished;
     }
 
-    /**
-     * @return string
-     */
-    public function getState()
+    public function getState(): string
     {
         return $this->state;
     }
 
-    /**
-     * @return string
-     */
-    public function getStartTime()
+    public function getStartTime(): float
     {
         return $this->startTime;
     }
 
-    /**
-     * @return string
-     */
-    public function getEndTime()
+    public function getEndTime(): float
     {
         return $this->endTime;
     }
 
     /**
-     * @return string
-     *
      * @deprecated
      */
-    public function getPlaybackType()
+    public function getPlaybackType(): string
     {
         return $this->playbackType;
     }
 
     /**
-     * @return string
-     *
      * @deprecated
      */
-    public function getPlaybackUrl()
+    public function getPlaybackUrl(): string
     {
         return $this->playbackUrl;
     }
 
     /**
-     * @return string
-     *
      * @deprecated
      */
-    public function getPlaybackLength()
+    public function getPlaybackLength(): int
     {
         return $this->playbackLength;
     }
 
     /**
-     * @return array
+     * @return array<string, string>
      */
-    public function getMetas()
+    public function getMetas(): array
     {
         return $this->metas;
     }
 
     /**
      * @return Format[]
      */
-    public function getFormats()
+    public function getFormats(): array
     {
-        if (null === $this->formats) {
-            $this->formats = [];
-            foreach ($this->rawXml->playback->format as $formatXml) {
-                $this->formats[] = new Format($formatXml);
+        $formats = [];
```

Comment:
Trade-off between more memory (old version) and more cpu (new version). Need a good argument to decided which solution to choose for the community. We assume the XML content is static and never changes in a single API call.

Reply:
Technically you are right (memory vs. cpu), but I guess those API-small data do no harm to any of them in these days. This proposal is just a cleaner and leaner way ... and it is more robust against programming errors (less code, less variables, low risk of inconsistent data).